src/libsyncengine/update_detection/update_detector/node.cpp

```diff
@@ -179,16 +180,25 @@ SyncPath Node::getPath() const {
     return path;
 }
 
-bool Node::isParentValid(std::shared_ptr<const Node> parentNode) const {
-    if (!parentNode) return true; // `parentNode` is the root node, hence a valid parent. Stop climbing up the tree.
-
-    while (parentNode) {
-        if (parentNode->id() == _id)
-            return false; // This node is a parent of `parentNode`, hence `parentNode` is not a valid parent.
-        parentNode = parentNode->parentNode();
+bool Node::isParentOf(std::shared_ptr<const Node> potentialChild) const {
+    if (!potentialChild) return false; // `parentNode` is the root node, 
+    if (potentialChild->id().has_value() && potentialChild->id() == _id) return false; // potentialChild cannot be its own parent
+    while (potentialChild) {
+        if (!potentialChild->id().has_value()) {
+            LOG_ERROR(Log::instance()->getLogger(), "Error in Node::isParentOf: Node has no id.");
+            assert(false && "Node has no id");
+            return false;
+        }
+        if (*potentialChild->id() == *_id)
+            return true; // This node is a parent of `potentialChild`
+        potentialChild = potentialChild->parentNode();
     }
 
-    return true;
+    return false;
+}
+
+bool Node::isParentValid(std::shared_ptr<const Node> parentNode) const {
```

Comment:
This sounds off... not being the child of a node does not necessary means being the parent of the particular node.
Take the following example:
.
├── A/
│   ├── AA
│   └── AB
└── B/
    ├── BA
    └── BB

`nodeA.isParentValid(nodeBA)` will return `true` because `BA` is not a child of `A`. But `A` is not a parent of `BA` either.

Reply:
You are right. The name is a bit confusing, but it was already the case before you PR and I do not have anything better yet.